File: src/3099.harshad-number.3371/solution.ts
function sumOfTheDigitsOfHarshadNumber (x: number): number {
    let n = x;
    let sum = 0;
    while (n > 0) {
        const digi = n % 10;
        sum += digi;
        n = (n - digi) / 10;
    }
    if (x % sum === 0) {
        return sum;
    }
    return -1;
}

File: src/3099.harshad-number.3371/solution.java
class Solution {
    public int sumOfTheDigitsOfHarshadNumber(int x) {
        int n = harshad(x);
        if(x%n == 0){
            return n;
        }
        return -1;
    }

    private int harshad(int x){
        int sum = 0;
        while (x != 0){
            sum += x%10;
            x /= 10;
        }
        return sum;
    }
}